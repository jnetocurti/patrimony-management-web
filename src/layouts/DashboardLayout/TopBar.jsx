import React, { useState } from 'react';
import { Link as RouterLink } from 'react-router-dom';

import {
  AppBar,
  Badge,
  Box,
  Hidden,
  IconButton,
  Toolbar,
  makeStyles
} from '@material-ui/core';
import InputIcon from '@material-ui/icons/Input';
import MenuIcon from '@material-ui/icons/Menu';
import NotificationsIcon from '@material-ui/icons/NotificationsOutlined';

import clsx from 'clsx';
import PropTypes from 'prop-types';
import Logo from 'src/components/Logo';

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.background.default
  },
  avatar: {
    width: 60,
    height: 60
  }
}));

const TopBar = ({ className, onMobileNavOpen, ...rest }) => {
  const classes = useStyles();
  const [notifications] = useState([]);

  return (
    <AppBar className={clsx(classes.root, className)} elevation={0} {...rest}>
      <Toolbar>
        <RouterLink to="/">
          <Logo />
        </RouterLink>
        <Box flexGrow={1} />
        <Hidden mdDown>
          <IconButton color="primary">
            <Badge
              badgeContent={notifications.length}
              color="default"
              variant="dot"
            >
              <NotificationsIcon />
            </Badge>
          </IconButton>
          <IconButton color="primary">
            <InputIcon />
          </IconButton>
        </Hidden>
        <Hidden lgUp>
          <IconButton color="default" onClick={onMobileNavOpen}>
            <MenuIcon />
          </IconButton>
        </Hidden>
      </Toolbar>
    </AppBar>
  );
};

TopBar.propTypes = {
  className: PropTypes.string,
  onMobileNavOpen: PropTypes.func
};

export default TopBar;
