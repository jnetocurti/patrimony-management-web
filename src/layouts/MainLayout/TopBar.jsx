import React from 'react';
import { Link as RouterLink } from 'react-router-dom';

import { AppBar, Toolbar, makeStyles } from '@material-ui/core';

import clsx from 'clsx';
import PropTypes from 'prop-types';
import Logo from 'src/components/Logo';

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.background.default
  },
  toolbar: {
    height: 64
  }
}));

const TopBar = ({ className, ...rest }) => {
  const classes = useStyles();

  return (
    <AppBar className={clsx(classes.root, className)} elevation={0} {...rest}>
      <Toolbar className={classes.toolbar}>
        <RouterLink to="/">
          <Logo />
        </RouterLink>
      </Toolbar>
    </AppBar>
  );
};

TopBar.propTypes = {
  className: PropTypes.string
};

export default TopBar;
