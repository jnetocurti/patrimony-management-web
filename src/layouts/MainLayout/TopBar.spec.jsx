import React from 'react';

import { shallow } from 'enzyme';

import TopBar from './TopBar';

describe('src/layouts/MainLayout/TopBar', () => {
  it('should render correctly', () => {
    const component = shallow(<TopBar />);
    expect(component).toMatchSnapshot();
  });
});
