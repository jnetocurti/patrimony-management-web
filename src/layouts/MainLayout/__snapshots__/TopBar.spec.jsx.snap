// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/layouts/MainLayout/TopBar should render correctly 1`] = `
<WithStyles(ForwardRef(AppBar))
  className="makeStyles-root-1"
  elevation={0}
>
  <WithStyles(ForwardRef(Toolbar))
    className="makeStyles-toolbar-2"
  >
    <Link
      to="/"
    >
      <Logo />
    </Link>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
